cosmos/providers/dbt/core/operators_kubernetes.py

```diff
@@ -0,0 +1,389 @@
+        self.selector = selector
+        self.vars = vars
+        self.models = models
+        self.cache_selected_only = cache_selected_only
+        self.no_version_check = no_version_check
+        self.fail_fast = fail_fast
+        self.quiet = quiet
+        self.warn_error = warn_error
+        self.db_name = db_name
+        self.schema = schema
+        self.env = env
+        self.append_env = append_env
+        self.output_encoding = output_encoding
+        self.skip_exit_code = skip_exit_code
+        self.cancel_query_on_kill = cancel_query_on_kill
+        self.dbt_executable_path = dbt_executable_path
+        self.container_flags = container_flags
+        super().__init__(**kwargs)
+
+    @cached_property
+    def subprocess_hook(self):
+        """Returns hook for running the bash command."""
+        return SubprocessHook()
+
+    def get_env(self, context):
+        """Builds the set of environment variables to be exposed for the bash command."""
+        system_env = os.environ.copy()
+        env = self.env
+        if env is None:
+            env = system_env
+        else:
+            if self.append_env:
+                system_env.update(env)
+                env = system_env
+
+        airflow_context_vars = context_to_airflow_vars(context, in_env_var_format=True)
+        self.log.debug(
+            "Exporting the following env vars:\n%s",
+            "\n".join(f"{k}={v}" for k, v in airflow_context_vars.items()),
+        )
+        env.update(airflow_context_vars)
+
+        return env
+
+    def exception_handling(self, result):
+        if self.skip_exit_code is not None and result.exit_code == self.skip_exit_code:
+            raise AirflowSkipException(
+                f"dbt command returned exit code {self.skip_exit_code}. Skipping."
+            )
+        elif result.exit_code != 0:
+            raise AirflowException(
+                f"dbt command failed. The command returned a non-zero exit code {result.exit_code}."
+            )
+
+    def add_global_flags(self):
+        global_flags = [
+            "project_dir",
+            "profiles_dir",
+            "select",
+            "exclude",
+            "selector",
+            "vars",
+            "models",
+        ]
+
+        flags = []
+        for global_flag in global_flags:
+            dbt_name = f"--{global_flag.replace('_', '-')}"
+
+            global_flag_value = self.container_flags.get(global_flag)
+            if global_flag_value is None:
+                global_flag_value = self.__getattribute__(global_flag)
+
+            if global_flag_value is not None:
+                if isinstance(global_flag_value, dict):
+                    # handle dict
+                    yaml_string = yaml.dump(global_flag_value)
+                    flags.append(dbt_name)
+                    flags.append(yaml_string)
+                else:
+                    flags.append(dbt_name)
+                    flags.append(str(global_flag_value))
+
```

Comment:
Same here. Maybe there's a class these two classes can inherit for these shared functions.

Reply:
It's done, I created a super class used by all operators to factorize the common code.